benchmarking/smartseq2/R/analysis_functions.R

```diff
@@ -0,0 +1,327 @@
+  return(list('p' = density.p))
+}
+# ks test
+plotKS <- function(df) {
+  mz <- melt(df)
+  x <- df$Base
+  y <- df$Updated
+  ks <- ks.test(x, y)
+  # run cdf
+  cdf1 <- ecdf(x)
+  cdf2 <- ecdf(y)
+  # find max distance D-stats
+  minMax <- seq(min(x, y), max(x, y), length.out = length(x))
+  x0 <-
+    minMax[which(abs(cdf1(minMax) - cdf2(minMax)) == max(abs(cdf1(minMax) - cdf2(minMax))))]
+  y0 <- cdf1(x0)
+  y1 <- cdf2(x0)
+  # plot ecdf
+  ks.p <-
+    ggplot(mz, aes(x = value, group = variable, color = variable)) +
+    stat_ecdf(size = 1) +
+    theme_bw() +
+    theme(legend.position = "top") +
+    ylab("ECDF")
+  # do KS test
+  dtable <-
+    data.frame(
+      'test' = 'K-S',
+      'D-stats' = round(ks$statistic, 4),
+      'P-value' = convert2star(ks$p.value)
+    )
+  dtable.p <-
+    ggtexttable(dtable,
+                row = NULL,
+                theme = ttheme('mBlue', base_size = 18))
+  # plot segement to represent the D statistics
+  ks.p <-
+    ks.p + geom_segment(aes(
+      x = x0[1],
+      y = y0[1],
+      xend = x0[1],
+      yend = y1[1]
+    ),
+    linetype = "dashed",
+    color = "red")
+  ks.p <-
+    ks.p + geom_point(aes(x = x0[1] , y = y0[1]), color = "red", size = 4)
+  ks.p <-
+    ks.p + geom_point(aes(x = x0[1] , y = y1[1]), color = "red", size = 4)
+  ks.p <- ks.p + ggtitle(paste("K-S Test"))
+  ks.p <- addTheme(ks.p)
+  ks.p <-
+    ks.p + theme(legend.title = element_blank()) + annotation_custom(ggplotGrob(dtable.p))
+  return(list(
+    'p' = ks.p,
+    'D' = ks$statistic,
+    'pval' = ks$p.value
+  ))
+}
+# parse GTF file
+ParseGene <- function(gtf_file) {
+  gtf_gencode <-
+    readGFF(
+      gtf_file,
+      version = 2L,
+      tags = c("gene_name", "gene_id", "transcript_id", "gene_type")
+    )
+  genes <- subset(gtf_gencode, gtf_gencode$type == "gene")
+  return(genes)
+}
+# cover to foldchnages
+# matrix1 and matrix2 are two data matrix
+foldchanges <- function(mat1, mat2) {
+  # log2 transformation
+  # first 2 columns are gene ID and length
+  logmd1 <- log(mat1[,-c(1:2)] + 1, base = 2)
+  # match gene ID
+  mlist <- match(mat1[, 1], mat2[, 1])
+  # match sample column
+  nlist <- match(colnames(mat1), colnames(mat2))
+  mat2 <- mat2[mlist, nlist]
+  # log2 transformation
+  logmd2 <- log(mat2[mlist,-c(1:2)] + 1, base = 2)
+  fcmat <- logmd1 - logmd2
+  out <- cbind(mat1[, 1:2], fcmat)
+  return(out)
+}
+# log2 transformation
+takelog2 <- function(mat) {
+  dd <- log(mat[, -c(1:2)] + 1, base = 2)
+  out <- cbind(mat[, c(1:2)], dd)
+  return(out)
+}
+RunCorrTest <- function(x, y) {
+  pval <- c()
+  cval <- c()
+  for (i in colnames(x)) {
+    z <- cor.test(x[, i], y[, i])
+    pval <- c(pval, z$p.value)
+    cval <- c(cval, z$estimate)
+  }
+  return(data.frame(
+    'pvalue' = pval,
+    'cor' = cval,
+    'sample' = colnames(x)
+  ))
+}
+# calculate correlation matrix btw matrix data
+CorrDataMatrix <- function(mat1, mat2, islog2) {
+  # match gene ID
+  if (islog2 == 1) {
```

Comment:
Can you make the changes from the comments about this function in #94 here as well?

Reply:
yes, I am changing  this for #94.